lib/image_info/parser.rb

```diff
@@ -26,6 +26,12 @@ def set_image_size
     def set_image_type
       image.type = parser.format
     end
-
+    
+    private
+    
+    def parser
+      @parser ||= ::ImageSize.new(@data) 
+    rescue ImageSize::FormatError
```

Comment:
you'll run into `nil` exceptions if this raise since you won't have a `parser` anymore